### ISSUE TITLE
Update tests for analyzer reachability warnings

### ISSUE THREAD

**Issue by munificent** [dart-lang/co19#1978]
I'm [updating the test runner](https://dart-review.googlesource.com/c/sdk/+/291967) so that we can write static error tests of unreachable case warnings (well, hints currently) from analyzer. Before this change, there's no way to write a language or co19 test that validates that analyzer reports unreachable case warnings.

This is, unsurprisingly, breaking a few tests. I've updated the ones in the Dart repo, but there are some co19 ones too:

* co19/Language/Statements/Switch/execution_case_no_default_t01
* co19/Language/Statements/Switch/execution_case_t01
* co19/LanguageFeatures/Patterns/cast_A01_t02
* co19/LanguageFeatures/Patterns/constant_A02_t15
* co19/LanguageFeatures/Patterns/constant_A02_t18
* co19/LanguageFeatures/Patterns/logical_or_A05_t05
* co19/LanguageFeatures/Patterns/null_assert_A03_t01
* co19/LanguageFeatures/Patterns/null_assert_A03_t02
* co19/LanguageFeatures/Patterns/null_check_A02_t01
* co19/LanguageFeatures/Patterns/object_A03_t01
* co19/LanguageFeatures/Patterns/object_A07_t05
* co19/LanguageFeatures/Patterns/pattern_context_A07_t02
* co19/LanguageFeatures/Patterns/pattern_variable_sets_A01_t03
* co19/LanguageFeatures/Patterns/pattern_variable_sets_A02_t01
* co19/LanguageFeatures/Patterns/scope_A04_t01
* co19/LanguageFeatures/Patterns/scope_A04_t02
* co19/LanguageFeatures/Patterns/shared_case_scope_A01_t01
* co19/LanguageFeatures/Patterns/shared_case_scope_A01_t02
* co19/LanguageFeatures/Patterns/shared_case_scope_A01_t03
* co19/LanguageFeatures/Patterns/switch_statement_A01_t01
* co19/LanguageFeatures/Patterns/type_inference_A25_t01
* co19/LanguageFeatures/Patterns/type_inference_A31_t01
* co19/LanguageFeatures/Patterns/variable_A03_t02
* co19/LanguageFeatures/Patterns/variable_A03_t03
* co19/LanguageFeatures/Patterns/variable_A03_t05

The linked CL should give some guidance on how to fix them. It's probably some combination of:

* If it's already a static error test, then go ahead and add expectations for the unreachability errors too. They look like:

    ```dart
        case A(a: Enum.a, b: false): // Unreachable
    //  ^^^^
    // [analyzer] HINT.UNREACHABLE_SWITCH_CASE
    ``` 

* If the cases are intended to be unrelated and are just put into a single switch for convenience, it may be easiest to split them into separate switch statements to make the errors go away. That's what I did with invalid_const_pattern_test.dart.

Let me know if you have any questions or if there's anything I can do to help. As always, thanks. :)

**Comment by sgrekhov** [comment]
@munificent is there a way to ignore this hint somehow? Sometimes, for example, we are testing that first matched case is executed first. In case of constant pattern it looks like
```dart
String test(int i) => switch (i) {
  1 => "case-1",
  1 => "case-2",
  _ => "no match"
}

main() {
  Expect.equals("case-1", test(1));
}
```
If to add expected hint, that this test won't be run at runtime. I'd be nice to have some flag, for such cases, like `// @ignore-hints` or something like this (`// SharedOptions=--ignore-hints` may be?). In this cases analyzer would ignore the hint and the runtime would tests what we are going to check

**Comment by munificent** [comment]
There was a bug in my CL which I just fixed. With that fix, the test runner understands that these unreachable hints are only warnings. This means that a test can be *both* a static error test and a runtime test. So if you put the unreachable case warning comments in there, the test runner will validate that analyzer reports them. But it will also still allow that test to be run on the VM and other configurations because the only static error markers in there are for non-fatal warnings.

So if you just put the comment markers in these tests, it should do what you want. (Or at least it will once my change lands. I've been fighting with the commit queue.)
